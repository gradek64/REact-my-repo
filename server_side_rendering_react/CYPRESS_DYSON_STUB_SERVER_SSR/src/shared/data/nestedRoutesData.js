const nestedRoutesData = [
  {
    description: "how the <Link> tag works ?",
    topicRoute: "linkTag",
    resources: {
      name: "<Link> tag",
      description:
        "Link tag it will always work on client and does not reload the page",
    },
  },
  {
    description: "how the the <a> tags work ?",
    topicRoute: "aTag",
    resources: {
      name: "<a> tag",
      description: "a tag will always reload the page causing server to re-run",
    },
  },
  {
    description: "const {url} =useRouteMatch() hook",
    topicRoute: "read-url-parameters",
    resources: {
      name: "const {url} = useRouteMatch()",
      description:
        "{url} used for nested links, cause ti will return something as: topics/react-router",
    },
  },
  {
    description: "const {path} =useRouteMatch() hook",
    topicRoute: "read-path-parameters",
    resources: {
      name: "const {path} = useRouteMatch()",
      description:
        "{path} used for nested routes, cause it will retain param as  path/:topicId",
    },
  },
  {
    description: "const {topicId} = useParams() hook",
    topicRoute: "read-params",
    resources: {
      name: "const {topicId} = useParams()",
      description: "used for getting /:topicId param ",
    },
  },
];

export default nestedRoutesData;
