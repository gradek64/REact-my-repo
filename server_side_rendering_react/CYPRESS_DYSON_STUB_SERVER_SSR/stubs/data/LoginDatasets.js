/* eslint-disable no-param-reassign */
// Login Datasets
function setupDefault(loginService) {
  loginService.setLogin('Default', 200, [], 0)
}
function setupMissingEmail(loginService) {
  loginService.setLogin('MissingEmail', 400, ['MISSING_EMAIL'])
}
function setupMissingPassword(loginService) {
  loginService.setLogin('MissingPassword', 400, ['MISSING_PASSWORD'])
}
function setupMissingCredentials(loginService) {
  loginService.setLogin('MissingCredentials', 400, ['MISSING_EMAIL', 'MISSING_PASSWORD'])
}
function setupInvalidEmail(loginService) {
  loginService.setLogin('InvalidEmail', 400, ['INVALID_EMAIL'])
}
function setupInvalidPassword(loginService) {
  loginService.setLogin('InvalidPassword', 400, ['INVALID_PASSWORD'])
}
function setupInvalidCredentials(loginService) {
  loginService.setLogin('InvalidCredentials', 400, ['INVALID_EMAIL', 'INVALID_PASSWORD'])
}
function setupMissingOrigin(loginService) {
  loginService.setLogin('MissingOrigin', 400, ['MISSING_ORIGIN'])
}
function setupAuthenticationError(loginService) {
  loginService.setLogin('AuthenticationError', 401)
}
function setupInternalServerError(loginService) {
  loginService.setLogin('InternalServerError', 500)
}
function setupRequestTimeout(loginService) {
  loginService.setLogin('RequestTimeout', 500)
}
function setupUiTimeout(loginService) {
  loginService.setLogin('UiTimeout', null)
}
function setupJsError(loginService) {
  loginService.setLogin('JsError', null)
}
function setupParseError(loginService) {
  loginService.setLogin('ParseError', null)
}

const loginDatasets = {
  Default: {
    description: '200: Successful login',
    info: 'E-mail and password match an existing Argos account',
    setup: setupDefault,
  },
  MissingEmail: {
    description: '400: Missing Email',
    info: 'No email has been submitted',
    setup: setupMissingEmail,
  },
  MissingPassword: {
    description: '400: Missing Password',
    info: 'No password has been submitted',
    setup: setupMissingPassword
  },
  MissingCredentials: {
    description: '400: Missing Credentials',
    info: 'Both email and password have not been submitted',
    setup: setupMissingCredentials
  },
  InvalidEmail: {
    description: '400: Invalid Email',
    info: 'Invalid email format',
    setup: setupInvalidEmail
  },
  InvalidPassword: {
    description: '400: Invalid Password',
    info: 'Invalid password format',
    setup: setupInvalidPassword
  },
  InvalidCredentials: {
    description: '400: Invalid Credentials',
    info: 'Both email and password have not the correct format',
    setup: setupInvalidCredentials
  },
  MissingOrigin: {
    description: '400: Missing Origin',
    info: 'No origin has been submitted',
    setup: setupMissingOrigin
  },
  AuthenticationError: {
    description: '401: AuthenticationError',
    info: 'E-mail and password don t match an existing Argos account',
    setup: setupAuthenticationError
  },
  InternalServerError: {
    description: '500: Internal Server Error',
    info: 'Internal Server Error',
    setup: setupInternalServerError
  },
  RequestTimeout: {
    description: '500: Request Timeout',
    info: 'The request timed out',
    setup: setupRequestTimeout
  },
  UiTimeout: {
    description: 'UI call handler time out',
    info: 'API failed to respond within XHR handler\'s wait limit',
    setup: setupUiTimeout
  },
  JsError: {
    description: 'UI encounters a JavaScript error',
    info: 'An error occurred in the UI',
    setup: setupJsError
  },
  ParseError: {
    description: 'Error parsing response',
    info: 'A malformed JSON response causes a parsing error',
    setup: setupParseError
  }
}

module.exports = loginDatasets
