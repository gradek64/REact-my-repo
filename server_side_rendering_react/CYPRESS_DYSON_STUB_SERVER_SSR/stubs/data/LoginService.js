const {
  DEFAULT_RESPONSE_COOKIE,
  DEFAULT_LOGIN_RESPONSE,
  DEFAULT_REGISTRATION_RESPONSE,
  DEFAULT_FORGOTTEN_PASSWORD_RESPONSE,
  DEFAULT_PASSWORD_LINK_RESPONSE,
  DEFAULT_PASSWORD_CHANGE_RESPONSE
} = require('./constants')

class LoginService {
  setDefaults() {
    this.setCookie(DEFAULT_RESPONSE_COOKIE)
    this.setLogin(DEFAULT_LOGIN_RESPONSE, 200, [], 0)
    this.setRegistration(DEFAULT_REGISTRATION_RESPONSE, 200)
    this.setForgottenPassword(DEFAULT_FORGOTTEN_PASSWORD_RESPONSE, 200)
    this.setPasswordLink(DEFAULT_PASSWORD_LINK_RESPONSE, 200)
    this.setPasswordChange(DEFAULT_PASSWORD_CHANGE_RESPONSE, 200, [])
  }

  constructor() {
    this.setDefaults()
  }

  toJson() {
    return {
      responseCookie: this.responseCookie,
      login: {
        type: this.login.type,
        authenticationRetry: this.login.authenticationRetry
      },
      registration: {
        type: this.registration.type
      },
      forgottenPassword: {
        type: this.forgottenPassword.type
      },
      passwordLink: {
        type: this.passwordLink.type
      },
      passwordChange: {
        type: this.passwordChange.type
      }
    }
  }

  fromJson(json) {
    this.setCookie((typeof json.responseCookie !== 'undefined' && String(json.responseCookie)) || DEFAULT_RESPONSE_COOKIE)
    this.login.type = (json.login && json.login.type) || DEFAULT_LOGIN_RESPONSE
    this.login.authenticationRetry = (json.login && json.login.authenticationRetry) || 0
    this.registration.type = (json.registration && json.registration.type) || DEFAULT_REGISTRATION_RESPONSE
    this.forgottenPassword.type = (json.forgottenPassword && json.forgottenPassword.type) || DEFAULT_FORGOTTEN_PASSWORD_RESPONSE
    this.passwordLink.type = (json.passwordLink && json.passwordLink.type) || DEFAULT_PASSWORD_LINK_RESPONSE
    this.passwordChange.type = (json.passwordChange && json.passwordChange.type) || DEFAULT_PASSWORD_CHANGE_RESPONSE
  }

  reset() {
    this.setDefaults()
  }

  setCookie(stringValue) {
    this.responseCookie = stringValue === 'true'
  }

  setLogin(type, status, errorCodes = [], authenticationRetry = null) {
    if (!this.login) {
      this.login = {}
    }
    this.login.type = type
    this.login.status = status
    this.login.errorCodes = errorCodes
    if (authenticationRetry !== null) {
      this.login.authenticationRetry = authenticationRetry
    }
  }

  setRegistration(type, status, errorCode = null) {
    if (!this.registration) {
      this.registration = {}
    }
    this.registration.type = type
    this.registration.status = status
    this.registration.errorCode = errorCode
  }

  setForgottenPassword(type, status, errorCode = null) {
    if (!this.forgottenPassword) {
      this.forgottenPassword = {}
    }
    this.forgottenPassword.type = type
    this.forgottenPassword.status = status
    this.forgottenPassword.errorCode = errorCode
  }

  setPasswordLink(type, status, errorCode = null) {
    if (!this.passwordLink) {
      this.passwordLink = {}
    }
    this.passwordLink.type = type
    this.passwordLink.status = status
    this.passwordLink.errorCode = errorCode
  }

  setPasswordChange(type, status, errorCodes = []) {
    if (!this.passwordChange) {
      this.passwordChange = {}
    }
    this.passwordChange.type = type
    this.passwordChange.status = status
    this.passwordChange.errorCodes = errorCodes
  }
}

module.exports = LoginService
