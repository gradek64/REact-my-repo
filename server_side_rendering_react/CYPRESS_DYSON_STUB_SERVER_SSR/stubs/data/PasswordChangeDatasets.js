/* eslint-disable no-param-reassign */
// Password Change Datasets
function setupDefault(loginService) {
  loginService.setPasswordChange('Default', 200)
}
function setupInvalidLink(loginService) {
  loginService.setPasswordChange('InvalidLink', 400, ['INVALID_LINK'])
}
function setupMissingEmail(loginService) {
  loginService.setPasswordChange('MissingEmail', 400, ['MISSING_LOGONID'])
}
function setupMissingPassword(loginService) {
  loginService.setPasswordChange('MissingPassword', 400, ['MISSING_LOGONPASSWORD'])
}
function setupMissingVerifyPassword(loginService) {
  loginService.setPasswordChange('MissingVerifyPassword', 400, ['MISSING_LOGONPASSWORDVERIFY'])
}
function setupMissingFieldValues(loginService) {
  loginService.setPasswordChange('MissingFieldValues', 400, ['MISSING_LOGONPASSWORD', 'MISSING_LOGONPASSWORDVERIFY'])
}
function setupInvalidEmail(loginService) {
  loginService.setPasswordChange('InvalidEmail', 400, ['PASSWORDS_POLICY_VIOLATED'])
}
function setupInvalidPassword(loginService) {
  loginService.setPasswordChange('InvalidPassword', 400, ['PASSWORDS_POLICY_VIOLATED'])
}
function setupPasswordMismatch(loginService) {
  loginService.setPasswordChange('PasswordMismatch', 400, ['PASSWORDS_POLICY_VIOLATED'])
}
function setupInternalServerError(loginService) {
  loginService.setPasswordChange('InternalServerError', 500)
}
function setupRequestTimeout(loginService) {
  loginService.setPasswordChange('RequestTimeout', 500)
}
function setupUiTimeout(loginService) {
  loginService.setPasswordChange('UiTimeout', null)
}
function setupJsError(loginService) {
  loginService.setPasswordChange('JsError', null)
}
function setupParseError(loginService) {
  loginService.setPasswordChange('ParseError', null)
}

const passwordChangeDatasets = {
  Default: {
    description: '200: Successful Request',
    info: 'Password updated successfully',
    setup: setupDefault,
  },
  InvalidLink: {
    description: '400: Invalid Link',
    info: 'Invalid or expired link',
    setup: setupInvalidLink
  },
  MissingEmail: {
    description: '400: Missing Email',
    info: 'No email has been submitted',
    setup: setupMissingEmail,
  },
  MissingPassword: {
    description: '400: Missing Password',
    info: 'No password has been submitted',
    setup: setupMissingPassword
  },
  MissingVerifyPassword: {
    description: '400: Missing Verify Password',
    info: 'No verify password has been submitted',
    setup: setupMissingVerifyPassword
  },
  MissingFieldValues: {
    description: '400: Missing Field Values',
    info: 'Both password and verify password have not been submitted',
    setup: setupMissingFieldValues
  },
  InvalidEmail: {
    description: '400: Invalid Email',
    info: 'Invalid email format',
    setup: setupInvalidEmail
  },
  InvalidPassword: {
    description: '400: Invalid Password',
    info: 'Invalid password format',
    setup: setupInvalidPassword
  },
  PasswordMismatch: {
    description: '400: Password Mismatch',
    info: 'Verify password mismatch',
    setup: setupPasswordMismatch
  },
  InternalServerError: {
    description: '500: Internal Server Error',
    info: 'Internal Server Error',
    setup: setupInternalServerError
  },
  RequestTimeout: {
    description: '500: Request Timeout',
    info: 'The request timed out',
    setup: setupRequestTimeout
  },
  UiTimeout: {
    description: 'UI call handler time out',
    info: 'API failed to respond within XHR handler\'s wait limit',
    setup: setupUiTimeout
  },
  JsError: {
    description: 'UI encounters a JavaScript error',
    info: 'An error occurred in the UI',
    setup: setupJsError
  },
  ParseError: {
    description: 'Error parsing response',
    info: 'A malformed JSON response causes a parsing error',
    setup: setupParseError
  }
}

module.exports = passwordChangeDatasets
