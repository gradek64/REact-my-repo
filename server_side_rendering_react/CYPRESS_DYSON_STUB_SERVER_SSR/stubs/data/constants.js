const DEFAULT_RESPONSE_COOKIE = 'true'
const DEFAULT_LOGIN_RESPONSE = 'Default'
const DEFAULT_REGISTRATION_RESPONSE = 'Default'
const DEFAULT_FORGOTTEN_PASSWORD_RESPONSE = 'Default'
const DEFAULT_PASSWORD_LINK_RESPONSE = 'Default'
const DEFAULT_PASSWORD_CHANGE_RESPONSE = 'Default'

module.exports = {
  DEFAULT_RESPONSE_COOKIE,
  DEFAULT_LOGIN_RESPONSE,
  DEFAULT_REGISTRATION_RESPONSE,
  DEFAULT_FORGOTTEN_PASSWORD_RESPONSE,
  DEFAULT_PASSWORD_LINK_RESPONSE,
  DEFAULT_PASSWORD_CHANGE_RESPONSE
}
