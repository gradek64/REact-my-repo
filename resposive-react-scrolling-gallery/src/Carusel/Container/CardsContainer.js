import React, { Component } from 'react';
import PropTypes from 'prop-types';

// custom
import Card from '../Item/Card';
import data from '../../data/data_test';
import variables from './carusel-style.scss';

// css setup;
const {
	showFrontImages,
	itemWidth: width,
	itemWidthSmallDesktop,
	smallDesktop,
} = variables;
// class component
class Carusel extends Component {
	state = {
		allItems: null,
		index: 0,

		setTimer: this.props.timer,
		showFrontImagesNumber: showFrontImages,
		itemWidth: width,
	};

	componentDidMount = () => {
		// that would be async call;
		const { setTimer } = this.state;
		this.setState({ allItems: data.items });
		// if timer
		if (setTimer) this.moveItemsTimer();
		this.goLeft = true;

		//devision for next images set
		this.devision = 7 % showFrontImages;
		console.log('showFrontImages', showFrontImages);
		console.log('this.devision,this.devision', this.devision);

		// addListener is a listener for matchMedia goes back to IE10
		window
			.matchMedia(`(max-width: ${smallDesktop})`)
			.addListener(this.addMediaBreakPoint);
	};

	componentDidUnMount = () => {
		// remmeber to remove once the component is dismputed/destroyed
		window
			.matchMedia(`(max-width: ${smallDesktop})`)
			.removeListener(this.addMediaBreakPoint);
	};

	addMediaBreakPoint = (e) => {
		if (e.matches) {
			// If media query matches
			this.setState({ itemWidth: itemWidthSmallDesktop });
		} else {
			this.setState({ itemWidth: width });
		}
	};

	moveItemsTimer = () => {
		const { delay } = this.props;
		this.timer = setInterval(() => {
			const { index, allItems, setTimer } = this.state;
			!setTimer ? clearInterval(this.timer) : false;
			if (setTimer) {
				index !== 0 && !this.goLeft
					? (this.prevItem(), (this.goLeft = false))
					: (this.goLeft = true);

				index !== allItems.length - showFrontImages && this.goLeft
					? (this.nextItem(), (this.goLeft = true))
					: (this.goLeft = false);
			}
		}, delay);
	};

	nextItem = (e) => {
		// diable timer on manual click next;
		if (e) this.setState({ setTimer: false });
		const { index } = this.state;

		this.setState({
			index: index < 8 ? index + 1 : 1,
		});
	};

	prevItem = (e) => {
		// diable timer on manual click prev;
		if (e) this.setState({ setTimer: false });
		const { index } = this.state;
		this.setState({
			index: index - 1,
		});
	};

	render() {
		const { allItems, index, showFrontImagesNumber, itemWidth } = this.state;
		console.log('index', index);
		return (
			<div className="App">
				<button
					type="button"
					className="hideForMobile"
					onClick={this.nextItem}
					// in order to test disabled state all conditions have to be on the state for testing to set up
					/* 	disabled={
						allItems && index === allItems.length - showFrontImagesNumber
					} */
					// disabled={allItems && index === allItems.length}
				>
					Next
				</button>
				<button
					type="button"
					className="hideForMobile"
					onClick={this.prevItem}
					/* 	disabled={index === 0} */
				>
					Prev
				</button>

				<div
					className="carusel-box"
					style={{
						// transform: `translateX(-${index.index * (100 / properties.length)}%)`,
						flexDirection: index < 0 ? 'row-reverse' : 'row',
					}}
				>
					<ul
						className="cards-slider-inner"
						style={{
							// transform: `translateX(-${index.index * (100 / properties.length)}%)`,
							transform: `translateX(${(this.devision - index) * itemWidth}%)`,
						}}
					>
						{
							/* eslint-disable no-underscore-dangle */
							data.items
								? data.items.map((item) => <Card key={item._id} {...item} />)
								: null
						}
					</ul>
					<ul
						className="cards-slider-inner"
						style={{
							// transform: `translateX(-${index.index * (100 / properties.length)}%)`,
							transform: `translateX(${(this.devision - index) * itemWidth}%)`,
						}}
					>
						{
							/* eslint-disable no-underscore-dangle */
							data.items
								? data.items.map((item) => <Card key={item._id} {...item} />)
								: null
						}
					</ul>
				</div>
			</div>
		);
	}
}

Carusel.propTypes = {
	timer: PropTypes.bool,
	delay: PropTypes.number,
};
Carusel.defaultProps = {
	timer: false,
	delay: 2000,
};
export default Carusel;
