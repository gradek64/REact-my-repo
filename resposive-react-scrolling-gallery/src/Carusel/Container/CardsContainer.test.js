import React from 'react';
import { shallow } from 'enzyme';
import CardsContainer from './CardsContainer';
import data from '../../data/data';

// needed to be mocked to make enviroment for a windows.matchMedia
// import matchMedia from './_mock_media_listener';

const { items } = data;


/*
  FIXED DATA
  change if component changes
*/
const showFrontImages = 4;

const wrapper = shallow(<CardsContainer />);
// U need to give this value to state due to lag in loading this exernally;
// this is fixed and it has to be the same in component change to that will mess up your tests
wrapper.setState({ showFrontImagesNumber: showFrontImages });
test('renders correctly', () => {
  expect(wrapper).toMatchSnapshot();
});


test('increase index on click next', () => {
  const next = wrapper.find('button').first();
  // whole structure
  // console.log(wrapper.instance());
  // DOMe element structure
  // console.log(next.html());

  // get current state property;
  // console.log(wrapper.state().index);
  const currentIndex = wrapper.state().index;
  // click next
  next.simulate('click');
  expect(wrapper.state().index).toBe(currentIndex + 1);
});

test('decrease index on click prev', () => {
  const prev = wrapper.find('button').at(1);
  // set index to 5;
  wrapper.setState({ index: 5 });
  const currentIndex = wrapper.state().index;
  // click next
  prev.simulate('click');
  expect(wrapper.state().index).toBe(currentIndex - 1);
});

test('prev button disbale on index===0 and enabled if index!==0', () => {
  // set index to 0;
  wrapper.setState({ index: 0 });
  const prev = wrapper.find('button').at(1);
  // props of enzyme child
  // console.log(prev.props());
  expect(prev.props().disabled).toBe(true);
  // set index to 1;
  wrapper.setState({ index: 1 });
  const prevChange = wrapper.find('button').at(1);
  expect(prevChange.props().disabled).toBe(false);
});

test('next button disbale on index===items.length and enabled if index!==items.length', () => {
  // set index to 0 items are already there and showFrontImagesNumber as well no need to set up ;
  wrapper.setState({ index: 20 });
  const next = wrapper.find('button').first();
  // wrapper.update();
  expect(next.props().disabled).toBe(false);
  // change index with callback optional no diffrence I think
  wrapper.setState({
    index: items.length - wrapper.state().showFrontImagesNumber,
  },
  () => {
    // U DONT NEED TO UPDATE WRAPPER IT WONT SEE CHANGES
    // wrapper.update();
    const nextUpate = wrapper.find('button').first();
    expect(nextUpate.props().disabled).toBe(true);
  });
});

test('it start timer when props timer=true is set', () => {
  const timerSet = shallow(<CardsContainer timer />);
  expect(timerSet.state().setTimer).toBe(true);
});

test('should cancel timer once next button clicked with event', () => {
  const timerSet = shallow(<CardsContainer timer />);
  // click action on next
  timerSet.find('button').first().simulate('click', { e: 'set' });
  expect(timerSet.state().setTimer).toBe(false);
});

test('should cancel timer once prev button clicked with event', () => {
  const timerSet = shallow(<CardsContainer timer />);
  // click action on next
  timerSet.find('button').at(1).simulate('click', { e: 'set' });
  expect(timerSet.state().setTimer).toBe(false);
});
