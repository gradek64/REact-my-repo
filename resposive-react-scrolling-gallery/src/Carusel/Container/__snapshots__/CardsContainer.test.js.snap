// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="App"
>
  <button
    className="hideForMobile"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    Next
  </button>
  <button
    className="hideForMobile"
    disabled={true}
    onClick={[Function]}
    type="button"
  >
    Prev
  </button>
  <div
    className="carusel-box"
  >
    <ul
      className="cards-slider-inner"
      style={
        Object {
          "transform": "translateX(-NaN%)",
        }
      }
    >
      <Card
        _id="593e9297e17df20c4a237d42"
        address="914 Argyle Road Alley"
        bathrooms={2}
        bedrooms={2}
        carSpaces={2}
        city="Singer"
        index={0}
        key="593e9297e17df20c4a237d42"
        latitude={-33.944576}
        longitude={151.25584}
        picture="images/tn_property01.jpg"
        price={937180}
      />
      <Card
        _id="593e9297ec4cca9c56bf61af"
        address="255 Raleigh Place"
        bathrooms={1}
        bedrooms={2}
        carSpaces={0}
        city="Machias"
        index={1}
        key="593e9297ec4cca9c56bf61af"
        latitude={-33.944471}
        longitude={151.2541}
        picture="images/tn_property02.jpg"
        price={703725}
      />
      <Card
        _id="593e929773c71925e5d7c11c"
        address="580 Amber Street"
        bathrooms={2}
        bedrooms={3}
        carSpaces={0}
        city="Bend"
        index={2}
        key="593e929773c71925e5d7c11c"
        latitude={-33.944644}
        longitude={151.24834}
        picture="images/tn_property03.jpg"
        price={837111}
      />
      <Card
        _id="593e92973d4d15eedb129aee"
        address="854 Varanda Place"
        bathrooms={2}
        bedrooms={3}
        carSpaces={0}
        city="Ivanhoe"
        index={3}
        key="593e92973d4d15eedb129aee"
        latitude={-33.940396}
        longitude={151.2469}
        picture="images/tn_property04.jpg"
        price={648223}
      />
      <Card
        _id="593e929728b7f8543b7907e1"
        address="857 Rockaway Parkway"
        bathrooms={1}
        bedrooms={2}
        carSpaces={1}
        city="Lydia"
        index={4}
        key="593e929728b7f8543b7907e1"
        latitude={-33.944562}
        longitude={151.2503}
        picture="images/tn_property05.jpg"
        price={771826}
      />
      <Card
        _id="593e92975996a6263d146444"
        address="543 Harman Street"
        bathrooms={1}
        bedrooms={2}
        carSpaces={1}
        city="Rosedale"
        index={5}
        key="593e92975996a6263d146444"
        latitude={-33.939304}
        longitude={151.25046}
        picture="images/tn_property01.jpg"
        price={686238}
      />
      <Card
        _id="593e9297a13e246da5ba0884"
        address="787 Gerald Court"
        bathrooms={2}
        bedrooms={2}
        carSpaces={2}
        city="Allison"
        index={6}
        key="593e9297a13e246da5ba0884"
        latitude={-33.937819}
        longitude={151.25774}
        picture="images/tn_property02.jpg"
        price={595377}
      />
      <Card
        _id="593e9297afa51a013fcd07d5"
        address="416 Vandervoort Avenue"
        bathrooms={1}
        bedrooms={4}
        carSpaces={1}
        city="Nicut"
        index={7}
        key="593e9297afa51a013fcd07d5"
        latitude={-33.943154}
        longitude={151.24687}
        picture="images/tn_property03.jpg"
        price={696189}
      />
      <Card
        _id="593e92972bd8fd05f2ab1a89"
        address="437 Mill Street"
        bathrooms={1}
        bedrooms={3}
        carSpaces={0}
        city="Canterwood"
        index={8}
        key="593e92972bd8fd05f2ab1a89"
        latitude={-33.943723}
        longitude={151.25278}
        picture="images/tn_property04.jpg"
        price={716114}
      />
      <Card
        _id="593e9297fe2e6d32c859ecf4"
        address="553 Friel Place"
        bathrooms={1}
        bedrooms={2}
        carSpaces={1}
        city="Cobbtown"
        index={9}
        key="593e9297fe2e6d32c859ecf4"
        latitude={-33.942459}
        longitude={151.24941}
        picture="images/tn_property05.jpg"
        price={739852}
      />
      <Card
        _id="593e92970481ccf52f948e6d"
        address="687 Opal Court"
        bathrooms={1}
        bedrooms={4}
        carSpaces={2}
        city="Blue"
        index={10}
        key="593e92970481ccf52f948e6d"
        latitude={-33.939606}
        longitude={151.25289}
        picture="images/tn_property01.jpg"
        price={625917}
      />
      <Card
        _id="593e9297e09417746b79e1c5"
        address="183 Canton Court"
        bathrooms={1}
        bedrooms={2}
        carSpaces={0}
        city="Craig"
        index={11}
        key="593e9297e09417746b79e1c5"
        latitude={-33.938557}
        longitude={151.24773}
        picture="images/tn_property02.jpg"
        price={591236}
      />
      <Card
        _id="593e92978e7160fa2b9992d9"
        address="234 Prospect Street"
        bathrooms={2}
        bedrooms={1}
        carSpaces={0}
        city="Orovada"
        index={12}
        key="593e92978e7160fa2b9992d9"
        latitude={-33.938345}
        longitude={151.25045}
        picture="images/tn_property03.jpg"
        price={995539}
      />
      <Card
        _id="593e92973459fbdfa1f1c041"
        address="205 Ocean Parkway"
        bathrooms={1}
        bedrooms={1}
        carSpaces={2}
        city="Clay"
        index={13}
        key="593e92973459fbdfa1f1c041"
        latitude={-33.937763}
        longitude={151.24721}
        picture="images/tn_property04.jpg"
        price={866924}
      />
      <Card
        _id="593e92970306538830483e14"
        address="324 Degraw Street"
        bathrooms={2}
        bedrooms={2}
        carSpaces={2}
        city="Windsor"
        index={14}
        key="593e92970306538830483e14"
        latitude={-33.938662}
        longitude={151.2513}
        picture="images/tn_property05.jpg"
        price={810745}
      />
      <Card
        _id="593e92976042982766c97ca1"
        address="644 Ashford Street"
        bathrooms={1}
        bedrooms={2}
        carSpaces={0}
        city="Bluetown"
        index={15}
        key="593e92976042982766c97ca1"
        latitude={-33.940989}
        longitude={151.25156}
        picture="images/tn_property01.jpg"
        price={793134}
      />
      <Card
        _id="593e92971901e888a036a258"
        address="472 Wilson Street"
        bathrooms={1}
        bedrooms={1}
        carSpaces={1}
        city="Wakulla"
        index={16}
        key="593e92971901e888a036a258"
        latitude={-33.943395}
        longitude={151.24723}
        picture="images/tn_property02.jpg"
        price={984589}
      />
      <Card
        _id="593e929780e6af9c11a10b90"
        address="455 Crown Street"
        bathrooms={2}
        bedrooms={1}
        carSpaces={0}
        city="Greer"
        index={17}
        key="593e929780e6af9c11a10b90"
        latitude={-33.941819}
        longitude={151.25784}
        picture="images/tn_property03.jpg"
        price={836244}
      />
      <Card
        _id="593e9297441033e3f818ff6b"
        address="459 Duffield Street"
        bathrooms={2}
        bedrooms={3}
        carSpaces={1}
        city="Wright"
        index={18}
        key="593e9297441033e3f818ff6b"
        latitude={-33.940486}
        longitude={151.25562}
        picture="images/tn_property04.jpg"
        price={962426}
      />
      <Card
        _id="593e9297964d25b59d94e8aa"
        address="537 Gelston Avenue"
        bathrooms={2}
        bedrooms={3}
        carSpaces={0}
        city="Manchester"
        index={19}
        key="593e9297964d25b59d94e8aa"
        latitude={-33.942826}
        longitude={151.24943}
        picture="images/tn_property05.jpg"
        price={768916}
      />
      <Card
        _id="593e929737636c879539416d"
        address="689 Orange Street"
        bathrooms={1}
        bedrooms={1}
        carSpaces={1}
        city="Ballico"
        index={20}
        key="593e929737636c879539416d"
        latitude={-33.943478}
        longitude={151.24779}
        picture="images/tn_property01.jpg"
        price={956974}
      />
      <Card
        _id="593e9297b2516c3e5af4601f"
        address="874 Rugby Road"
        bathrooms={1}
        bedrooms={2}
        carSpaces={2}
        city="Tuskahoma"
        index={21}
        key="593e9297b2516c3e5af4601f"
        latitude={-33.93817}
        longitude={151.25545}
        picture="images/tn_property02.jpg"
        price={875917}
      />
      <Card
        _id="593e92972f6034590df04c45"
        address="687 Alabama Avenue"
        bathrooms={1}
        bedrooms={3}
        carSpaces={2}
        city="Temperanceville"
        index={22}
        key="593e92972f6034590df04c45"
        latitude={-33.938195}
        longitude={151.25286}
        picture="images/tn_property03.jpg"
        price={913858}
      />
      <Card
        _id="593e9297aa0e233a421f1a58"
        address="739 Vermont Street"
        bathrooms={2}
        bedrooms={2}
        carSpaces={0}
        city="Cliffside"
        index={23}
        key="593e9297aa0e233a421f1a58"
        latitude={-33.94209}
        longitude={151.25645}
        picture="images/tn_property04.jpg"
        price={622995}
      />
      <Card
        _id="593e92977526a30b580dcfb2"
        address="735 Humboldt Street"
        bathrooms={2}
        bedrooms={2}
        carSpaces={1}
        city="Sunbury"
        index={24}
        key="593e92977526a30b580dcfb2"
        latitude={-33.944276}
        longitude={151.25442}
        picture="images/tn_property05.jpg"
        price={593777}
      />
      <Card
        _id="593e92975141c74b24375bd9"
        address="617 Charles Place"
        bathrooms={1}
        bedrooms={3}
        carSpaces={1}
        city="Gracey"
        index={25}
        key="593e92975141c74b24375bd9"
        latitude={-33.939279}
        longitude={151.25559}
        picture="images/tn_property01.jpg"
        price={757686}
      />
      <Card
        _id="593e9297b3ffeb95bb008453"
        address="786 George Street"
        bathrooms={1}
        bedrooms={3}
        carSpaces={0}
        city="Farmers"
        index={26}
        key="593e9297b3ffeb95bb008453"
        latitude={-33.943311}
        longitude={151.25781}
        picture="images/tn_property02.jpg"
        price={976878}
      />
      <Card
        _id="593e9297e262c91acd91ab81"
        address="711 Sunnyside Avenue"
        bathrooms={2}
        bedrooms={1}
        carSpaces={0}
        city="Mapletown"
        index={27}
        key="593e9297e262c91acd91ab81"
        latitude={-33.943475}
        longitude={151.24929}
        picture="images/tn_property03.jpg"
        price={804859}
      />
      <Card
        _id="593e9297df21e62e0465f91f"
        address="403 Lawrence Avenue"
        bathrooms={1}
        bedrooms={2}
        carSpaces={2}
        city="Snyderville"
        index={28}
        key="593e9297df21e62e0465f91f"
        latitude={-33.943007}
        longitude={151.25262}
        picture="images/tn_property04.jpg"
        price={611018}
      />
      <Card
        _id="593e929787f567dccda86910"
        address="265 Ocean Court"
        bathrooms={1}
        bedrooms={1}
        carSpaces={1}
        city="Inkerman"
        index={29}
        key="593e929787f567dccda86910"
        latitude={-33.943142}
        longitude={151.25074}
        picture="images/tn_property05.jpg"
        price={707939}
      />
    </ul>
  </div>
</div>
`;
